xskillscore/core/np_deterministic.py

```diff
@@ -73,6 +75,69 @@ def _pearson_r(a, b, weights, axis):
     return res
 
 
+def _spearman_r(a, b, weights, axis):
+    """
+    ndarray implementation of scipy.stats.spearmanr.
+
+    Parameters
+    ----------
+    a : ndarray
+        Input array.
+    b : ndarray
+        Input array.
+    axis : int
+        The axis to apply the correlation along.
+    weights : ndarray
+        Input array.
+
+    Returns
+    -------
+    res : ndarray
+        Spearmanr's correlation coefficient.
+
+    See Also
+    --------
+    scipy.stats.spearmanr
+
+    """
+    a = bn.rankdata(a, axis=axis)
+    b = bn.rankdata(b, axis=axis)
+    return _pearson_r(a, b, weights, axis)
+
+
+def _spearman_r_p_value(a, b, weights, axis):
+    """
+    ndarray implementation of scipy.stats.spearmanr.
+
+    Parameters
+    ----------
+    a : ndarray
+        Input array.
+    b : ndarray
+        Input array.
+    axis : int
+        The axis to apply the correlation along.
+    weights : ndarray
+        Input array.
+
+    Returns
+    -------
+    res : ndarray
+        2-tailed p-value.
+
+    See Also
+    --------
+    scipy.stats.spearmanr
+
+    """
+    # https://github.com/scipy/scipy/blob/v1.3.1/scipy/stats/stats.py#L3613-L3764
+    rs = _spearman_r(a, b, weights, axis)
+    dof = a.shape[axis] - 2  # degrees of freedom
```

Comment:
Maybe handle this as in the same way as `_pearson_r_p_value`:

```python
a = np.rollaxis(a, axis)
dof = a.shape[0] - 2
```

Reply:
Also I would either change this to `df` as in `_pearson_r_p_value` or change that one to `dof`. Probably the latter since `df` is like a pandas dataframe.